pyvista/core/utilities/transformations.py

```diff
@@ -392,3 +394,176 @@ def rotation(
         translate_from_origin[:3, 3] = valid_point
 
         return translate_from_origin @ rotate @ translate_to_origin
+
+
+def decompose(
+    transformation: TransformLike,
+    *,
+    homogeneous: bool = False,
+    allow_negative_scale: bool = False,
+) -> tuple[NumpyArray[float], NumpyArray[float], NumpyArray[float], NumpyArray[float]]:
+    """Decompose a transformation into its components.
+
+    The transformation matrix ``M`` is decomposed into four components:
+
+    - translation ``T``
+    - rotation ``R``
+    - scaling ``S``
+    - shearing ``K``
+
+    such that, when represented as 4x4 matrices, ``M = TRSK``.
+
+    Reflections are represented implicitly in the decomposition:
+
+    - When ``allow_negative_scale`` is ``False`` (default), reflections are included
+      with the rotation ``R``. The rotation is left-handed (negative determinant) if
+      there is a reflection in the decomposition, or right-handed (positive
+      determinant) if there is no reflection.
+
+    - When ``allow_negative_scale`` is ``True``, reflections are included with the
+      scaling ``S``. The first scaling factor is negative if there is a reflection
+      in the decomposition, and positive if there is no reflection. The second and third
+      scaling factors are always positive.
+
+    By default, compact representations of the transformations are returned (i.e. as a
+    3-element vector or a 3x3 matrix). Optionally, 4x4 matrices may be returned instead.
+
+    .. note::
+
+        - The decomposition is computed using polar decomposition.
+        - The decomposition is unique (up to a change in sign depending on whether
+          reflections are included in the rotation or scaling component).
+
+    Parameters
+    ----------
+    transformation : TransformLike
+        Array or transform to decompose.
+
+    homogeneous : bool, default: False
+        If ``True``, return the components (translation, rotation, etc.) as
+        4x4 matrices. By default, translation and scaling components are returned as
+        length-3 vectors, and rotation and shear components are returned as 3x3 matrices.
+
+    allow_negative_scale : bool, default: False
+        If ``True``, the first scaling term may be negative. By default, the
+        scaling factors are always positive.
```

Comment:
This is kind of left over from the previous ``transforms3d`` implementation. I think it might be an unnecessary complication. We can probably remove it, such that scaling factors returned by this decomposition are _always_ positive, and reflections (if any) are _always_ included with the rotation. This is also how the current polar decomposition works naturally by default, so this would also remove a processing step.

Reply:
The main thing this option does is multiply the reflection with either the rotation term (`allow_negative_scale=False`) or the scaling term (`allow_negative_scale=True`).

But, the naming is awkward, and makes it harder to check for reflections. Instead, I think it's best to simply decompose the reflection term explicitly. This was done in https://github.com/pyvista/pyvista/pull/6628/commits/ded9567ee37196a0e2a4f896b33d1e6d7baf3cfa.

So the decomposition is now `TRNSK` instead of `TRSK` with `N` as the reflection.